### ISSUE TITLE
[request] Image Mask bulk generation

### ISSUE THREAD

**Issue by natowi** [alicevision/meshroom#713]
Bulk generate image masks using detected features and image matching results

Here is an idea to automatize object masking to optimize this new feature https://github.com/alicevision/meshroom/pull/708: 

We know from ImageMatching the corresponding features in the other images.
How about selecting the Keypoints on our object (simple toolbox) in one image and then applying the selection to the other images (and allow adjustments).

Then we could use a foreground extraction algorithm like this:

http://opensource.graphics/how-to-code-a-nice-user-guided-foreground-extraction-algorithm/

http://opensource.graphics/how-to-code-a-nice-user-guided-foreground-extraction-algorithm-addendum/

So we could bulk-create masks with minimal user interaction.

Implementation details:

Once ImageMatching is complete, there a new icon in the Viewer could be activated to select features using a simple toolbox.

The Bulk masking tool/helper could be integrated in the main gui (no new node required),

- roughly mask few images before starting the computation and use this as input to select detected features
or
- create a new "user interaction required" node type for this to select the initial area before bulk create the selection masks.

Similar issues: #188 #701

**Comment by ChemicalXandco** [comment]
I am already working on a masking system based on [grab cut](https://docs.opencv.org/master/d8/d83/tutorial_py_grabcut.html) 

Why use features from the reconstruction? This makes no sense because it does not distinguish between background and foreground features.

**Comment by natowi** [comment]
> Why use features from the reconstruction? This makes no sense because it does not distinguish between background and foreground features.

I do not want to directly use the features for segmentation.
I want to (manually) select detected features in one image and transfer the selection to the other images, since we know from ImageMatching the images with corresponding features.

**Comment by fabiencastan** [comment]
Please notice that there is already a WIP development for image masking:
https://github.com/alicevision/meshroom/pull/708
https://github.com/alicevision/AliceVision/pull/715

**Comment by natowi** [comment]
@fabiencastan yes, but from my understanding this does not include the mask _generation_

**Comment by fabiencastan** [comment]
Yes, it does not include generic background removal, but it was just to mention it as this issue looks like the version 2 of this image masking node.

**Comment by MightyBOBcnc** [comment]
Double-DIP also does image segmentation of foreground and background but I do like natowi's idea as leveraging the information from ImageMatching does sound very robust as it wouldn't have to 're-train' itself for every camera angle.

https://github.com/yossigandelsman/DoubleDIP 
http://www.wisdom.weizmann.ac.il/~vision/DoubleDIP/index.html

**Comment by ALfuhrmann** [comment]
I implemented a simple background subtraction as a meshroom node, see  #188, [my comment](https://github.com/alicevision/meshroom/issues/188#issuecomment-566507568).

You just have to take one clean background image without the object. I added some parameters for dirty backgrounds, but with a clean one it works relatively stable.

![image](https://user-images.githubusercontent.com/3731183/70993220-d27bcc80-20cb-11ea-9e2c-2114cbe9f99d.png)

**Comment by natowi** [comment]
https://github.com/alicevision/meshroom/pull/750

**Comment by natowi** [comment]
https://github.com/alicevision/Meshroom/wiki/New-Features-in-Meshroom-2023.1#1-image-masking

**Comment by natowi** [comment]
Masking (image segmentation) is included in 2025.1 to be released within the next days.
